src/main/java/io/cdap/plugin/google/sheets/source/GoogleSheetsSourceConfig.java

```diff
@@ -228,16 +230,47 @@
   @Macro
   private String sheetFieldName;
 
+  public GoogleSheetsSourceConfig(String referenceName, @Nullable String sheetsIdentifiers, String formatting,
+                                  Boolean skipEmptyData, String columnNamesSelection,
+                                  @Nullable Integer customColumnNamesRow, String metadataFieldName,
+                                  Boolean extractMetadata, @Nullable Integer firstFooterRow,
+                                  @Nullable Integer lastHeaderRow, @Nullable Integer lastFooterRow,
+                                  @Nullable String lastDataColumn, @Nullable String lastDataRow,
+                                  @Nullable String metadataCells, @Nullable Integer readBufferSize,
+                                  @Nullable Boolean addNameFields, @Nullable String spreadsheetFieldName,
+                                  @Nullable String sheetFieldName) {
+
+    super(referenceName);
+    this.sheetsIdentifiers = sheetsIdentifiers;
+    this.formatting = formatting;
+    this.skipEmptyData = skipEmptyData;
+    this.metadataFieldName = metadataFieldName;
+    this.columnNamesSelection = columnNamesSelection;
+    this.customColumnNamesRow = customColumnNamesRow;
+    this.extractMetadata = extractMetadata;
+    this.firstFooterRow = firstFooterRow;
+    this.lastHeaderRow = lastHeaderRow;
+    this.lastDataColumn = lastDataColumn;
+    this.lastDataRow = lastDataRow;
+    this.lastFooterRow = lastFooterRow;
+    this.metadataCells = metadataCells;
+    this.readBufferSize = readBufferSize;
+    this.addNameFields = addNameFields;
+    this.spreadsheetFieldName = spreadsheetFieldName;
+    this.sheetFieldName = sheetFieldName;
+  }
+
+
+  public GoogleSheetsSourceConfig(String referenceName) {
+    super(referenceName);
+  }
+
   /**
    * Returns the instance of Schema.
    * @return The instance of Schema
    */
   public Schema getSchema() {
-    if (schema == null) {
-      if (dataSchemaInfo.isEmpty()) {
-        throw new RuntimeException("There are no headers to process. " +
```

Comment:
Is this not relevant anymore?

Reply:
It doesn't look relevant to me because we are already checking if Source folder (directory) contains any spreadsheets files in here: 

https://github.com/data-integrations/google-drive/blob/3074341092eecd2522ea5ce9ac1c40877e0fc03f/src/main/java/io/cdap/plugin/google/sheets/source/GoogleSheetsSourceConfig.java#L348-L354

...but I will revert this RuntimeException in getSchema().

Reply:
hm why are we throwing a RuntimeException here? If config validation fails, we should add the failure to the failure collector (pass in FailureCollector, call collector.addFailure()).

Reply:
PR updated.